File: rb/features.rb
# SatelliteTleData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module SatelliteTleDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      SatelliteTleDataBaseFeature.new
    when "ratelimit"
      SatelliteTleDataRatelimitFeature.new
    when "retry"
      SatelliteTleDataRetryFeature.new
    when "test"
      SatelliteTleDataTestFeature.new
    when "timeout"
      SatelliteTleDataTimeoutFeature.new
    else
      SatelliteTleDataBaseFeature.new
    end
  end
end

File: py/features.py
# SatelliteTleData SDK feature factory

from feature.base_feature import SatelliteTleDataBaseFeature
from feature.test_feature import SatelliteTleDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SatelliteTleDataBaseFeature(),
        "test": lambda: SatelliteTleDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
